### DISCUSSION TITLE
PrismaClientKnownRequestError: Invalid `prisma.b2bTransaction.findUnique()` invocation: The table `public.B2bTransaction` does not exist in the current database.

### DISCUSSION THREAD

**Discussion by pantharshit007** [prisma/prisma#25917] [Q&A] [answered]
### Bug description

prisma table `B2bTransaction` exists in database cross-checked withthe  studio also but when try to use in hono on production gives this error:

```
PrismaClientKnownRequestError: Invalid `prisma.b2bTransaction.findUnique()` invocation: The table `public.B2bTransaction` does not exist in the current database.
```

but, it works perfectly fine in local environment.

### How to reproduce

This is a production issue in hono, works fine in local environment.

### Expected behavior

It should work and perform the operation like it does in local environment

### Prisma information

```prisma
model B2bTransaction {
  id                 String            @id @default(uuid())
  amount             Int               
  timestamp          DateTime          @default(now())
  status             TransactionStatus
  type               TransactionType

  webhookId          String?          @unique  // To track webhook processing
  webhookStatus      WebhookStatus?
  webhookAttempts    Int              @default(0)
  lastWebhookAttempt DateTime?
  
  senderUserId       String?
  receiverUserId     String?
  
  senderAccountNumber    String?     
  receiverAccountNumber  String?     
  
  senderBankName        String
  receiverBankName      String
  
  
  senderUser         User?     @relation("SenderBankRelation", fields: [senderUserId], references: [id], onDelete: SetNull)
  receiverUser       User?     @relation("ReceiverBankRelation", fields: [receiverUserId], references: [id], onDelete: SetNull)
  
  @@index([senderUserId])
  @@index([receiverUserId])
  @@index([webhookId])
}
```

```ts

    if (!decryptedData.txnId) {
      return { success: false, message: "Transaction ID not found", paymentToken: null };
    }

    // fetch transaction data
    const transaction = await db.b2bTransaction.findUnique({
      where: {
        id: decryptedData.txnId,
        //webhookId: webhookId!
      },
      select: {
        id: true,
        senderUserId: true,
        receiverUserId: true,
        senderBankName: true,
        amount: true,
        status: true,
        webhookStatus: true,
      },
    });


```

### Environment & setup

- OS: Windows
- Database: PostgresSQL
- Node.js version: ^18

### Prisma Version

```
"@prisma/client": "^5.22.0",
```

**Comment by ludralph** [comment]
Hi @pantharshit007 

Did you deploy your database to production using the command `npx prisma migrate deploy`? Also, do you have the `prisma generate` command defined in your `postinstall` script in `package.json` file?

**Comment by pantharshit007** [comment]
> Hi @pantharshit007
> 
> Did you deploy your database to production using the command `npx prisma migrate deploy`? Also, do you have the `prisma generate` command defined in your `postinstall` script in `package.json` file?

Yeah as I have stated above I cross checked all the tables in studio and I am using `wrangler CLI` which deploys build to cloudflare directly

- I have done all `db push` , `db deploy`, `migrate`
- also have appropriate postinstall scripts in place 

If you want I can send you the link of repo

**Reply by ludralph** [reply]
Please do send the repo link. Thank you

**Reply by pantharshit007** [reply]
_Just a note:_ the current postinstall is configured for `vercel middleware`, I am deploying `bank-api` via wrangler CLI so no issue there

Thanks

**Reply by ludralph** [reply]
Thanks for sharing, I'll take a look and provide any updates I might have.

**Reply by pantharshit007** [reply, marked as answer]
hey @ludralph  closing this as the issue is resolved.

Just wanna say humans can be so stupid sometimes the problem was I have 2 URL's `DATABASE_URL` and `UNPOOLED_DATABASE_URL` so what was happening here is the database neon provided by default is there's `/neon` and I was using a different one for my project and somehow I copied `neon's` db in `DATABASE_URL` and `UNPOOLED_DATABASE_URL` is using my correct URL.

but locally I have 2 envs and the one which I am using locally for all shared packages has the correct `URL` but my `.dev.vars` got the wrong one which was propagating to Cloudflare.

Thanks
